File: 5.cpp
#include<bits/stdc++.h>
using namespace std;
string tostring(char c){
	stringstream stream;
	stream<<c;
	return stream.str();
}
class Solution {
public:
    static string longestPalindrome(string s) {
    	int answer = 0;
    	string answermid="";
    	string realanswer;
  		for (int i = 0;i<s.size();i++){
  			int now = 1;
  			int b = i;
  			int e = i;
  			answermid = tostring(s[i]); 
  			while (b-1>=0 && e+1<s.size() && s[b-1] == s[e+1]){
  				b = b-1;
				e = e+1;
				now++;now++;
				answermid = tostring(s[b])+answermid+tostring(s[e]);	
			}
			if (now>answer) {
			answer = now;
			realanswer = answermid;
			if (now == s.size()) return answermid;}
  			now = 1;
  			b = i;
  			e = i;
			if (b-1 >= 0 && s[b-1] == s[b]){
				answermid = tostring(s[b-1])+tostring(s[b]); 
				b = b -1;now++;
				while (b-1>=0 && e+1<s.size() && s[b-1] == s[e+1]){
  				b = b-1;
				e = e+1;
				now++;now++;	
				answermid  = tostring(s[b])+answermid+tostring(s[e]);
				}
			} 
			if (now > answer){
			answer = now;
			realanswer = answermid;}
			now = 1;
  			b = i;
  			e = i;
  			if (e+1<s.size() && s[e+1] == s[e]){
  				answermid = tostring(s[e+1])+tostring(s[e]);
  				e = e + 1;
				now++;	
				while (b-1>=0 && e+1<s.size() && s[b-1] == s[e+1]){
  				b = b-1;
				e = e+1;
				now++;now++;
				answermid  = tostring(s[b])+answermid+tostring(s[e]);
				}
			}
			if (now > answer) {
			answer = now;
			realanswer = answermid;}
		}
		return realanswer;    
    }
};

File: 5.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        answer = ""
        if (len(s) == 1):
            return s
        for i in range(len(s)-1):
            mid_answer_1 = s[i]
            st = i - 1
            ed = i + 1
            while st >= 0 and ed <= len(s)-1:
                    if (cmp(s[st],s[ed]) == 0):
                        mid_answer_1 = s[st] + mid_answer_1 + s[ed]
                    else:
                        break
                    st = st - 1
                    ed = ed + 1
            if (s[i] == s[i+1]):
                mid_answer_2 = s[i] + s[i+1]
                st = i - 1
                ed = i + 1 + 1
                while st >= 0 and ed <= len(s)-1:
                        if (s[st] == s[ed]):
                            mid_answer_2 = s[st] + mid_answer_2 + s[ed]
                        else:
                            break
                        st = st - 1
                        ed = ed + 1
                if (len(mid_answer_2) > len(answer)):
                    answer = mid_answer_2
            if (len(mid_answer_1) > len(answer)):
                answer = mid_answer_1
        return answer
if __name__ == '__main__':
    s = Solution()
    print(s.longestPalindrome("abcda"))
